modules/ROOT/pages/oauth-policy-implementation-concept.adoc

```diff
@@ -12,55 +12,85 @@ image::oauth-implementation-federated.png[]
 . The federated policy validates the token against the OAuth provider. To improve performance, Mule caches the call to the authorization server, which is only performed once during the lifetime of the token.
 +
 When OpenId, PingFederate, or OAuth 2.0 Access Token Enforcement Using External Provider policies are in effect and your API is deployed under Mule 4, the client ID returned by the OAuth provider is validated by the Client Id Enforcement policy.
-. All fulfilled requests in the flow are enriched with the information provided by the OAuth Provider when running under Mule 4, but information is not returned to the user. In releases earlier than Mule 4, this information was returned to the user.
-
-Possible responses are:
-
-*Mule 4.0.0 or later*
-
-* 200
-+
-The token was successfully validated.
+. As opposed to Mule 3, in Mule 4, all fulfilled requests in the flow are enriched with the information provided by the OAuth Provider, but this information is not returned to the user unless explicitly configured in the policy by enabling the *Expose Headers* checkbox:
 +
-* 400 Bad Request
+image::oauth-policy-implementation-concept-7ad82.png[]
 +
-This Bad Request error can be caused by: no token sent, the token was sent both as query param and header, or the token was sent as header but does not have prefix "Bearer".
+This flag blacks out `x-rate-limit` headers from the response. +
+The following example shows HTTP message headers returned as a token validation response.
 +
-* 401 Unauthorized
+----
+HTTP/1.1 200 OK
+Cache-Control: no-cache, no-store
+Date: Mon, 09 Mar 2015 19:08:07 GMT
+Accept-Ranges: bytes
+Server: Restlet-Framework/2.1.1
+Vary: Accept-Charset, Accept-Encoding, Accept-Language, Accept
+Content-Type: application/json;charset=UTF-8
+Transfer-Encoding: chunked
+{"uid":"john.doe","mail":"john.doe@example.com","scope":["uid","mail","cn","givenName"],"grant_type":"password","cn":"John Doe Full","realm":"/","token_type":"Bearer","expires_in":580,"givenName":"John","access_token":"fa017a0e-1bd5-214c-b19d-03efe9f9847e"}
+----
 +
-The token is unauthorized.
-+
-* 403 Forbidden
-+
-This error could be the result of the validation to the OAuth Provider, the scopes validation, or the result of Client ID Enforcement Policy.
+This token validation response may vary and, if *Expose Headers* is enabled, the proxy sends all headers received to the backend implementation with the exceptions of `scope` and `expires_in nodes`.
+
+Possible responses are:
+
+[%header%autowidth.spread,cols="a,a"]
+|===
+| Response | Reason
+| `200`
+| The token was successfully validated.
+| `400`
+| This Bad Request error can be caused by: no token sent, the token was sent both as query param and header, or the token was sent as header but does not have prefix "Bearer".
+| `401`
+| The token is unauthorized.
+| `403`
+| This error could be the result of the validation to the OAuth Provider, the scopes validation, or the result of Client ID Enforcement Policy.
```

Comment:
I think we need to avoid using "could" , Maybe this sentence can be changed to something like: 
"This response code can be caused by an error in the validation to the OAuth Provider, the scopes validation, or the Client ID Enforcement Policy."